src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp

```diff
@@ -407,6 +426,41 @@ void TransposedConvolution<
     aH = (outputHeight + kernelHeight - 2 * padHeight - 2) % strideHeight;
   }
 }
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename InputDataType,
+    typename OutputDataType
+>
+void TransposedConvolution<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    InputDataType,
+    OutputDataType
+>::InitializeSamePadding(){
+  /**
+   * Using O=s*(I-1) + K -2P + A
+   * where
+   * s=stride 
+   * I=Input Shape
+   * K=Kernel Size
+   * P=Padding
+   */
+  size_t totalPadWidth = (kernelWidth - strideWidth);
```

Comment:
It looks like the `output` would be similar to `input` only if `stride` is equal to one. 
This what actually TensorFlow does. However, we do not implement it in the similar way. 
We implement such that the input is always similar to the output irrespective of `stride`. So it would be great if you could implement it in the same way.

Reply:
For more info you can refer to the discussion in #2119.

Reply:
Hi @walragatver,
Yes I have implemented it like it is implemented in Tensorflow. I will make the changes to have output shape equal to input shape.

Reply:
I think I understand what needs to be done, I will commit required changes hopefully by tomorrow evening. Could you have a look then.

Reply:
I have made the changes such that output size is same as input size  however I am facing some problem while testing it in C++.
I tested the logic in python, 
`https://colab.research.google.com/drive/174HGBANY-cvnapDOeKrAT971rKx9jFYk`
and implemented the same here. I will try to fix it soon.

Reply:
@walragatver ,@zoq ,I have fixed the error now, I was missing a pair of parenthesis in the new expression.